commons-shared/src/main/scala/com/avsystem/commons/SharedExtensions.scala

```diff
@@ -206,27 +206,23 @@ object SharedExtensions extends SharedExtensions {
   }
 
   class CollectionOps[C[X] <: BTraversable[X], A](private val coll: C[A]) extends AnyVal {
-    def toMap[K, V](keyFun: A => K, valueFun: A => V): Map[K, V] = {
-      val res = Map.newBuilder[K, V]
-      coll.foreach { a =>
-        res += ((keyFun(a), valueFun(a)))
-      }
-      res.result()
+    def asMap[K, V](keyFun: A => K, valueFun: A => V): Map[K, V] = {
```

Comment:
I understand why this must be done, but `asMap` kind of suggest that this returns a view. Maybe `collectToMap` or `mkMap` would be better?